include/rpc/function_proxy.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <string>
+#include <future>
+
+#include "msgpack.hpp"
+
+namespace rpc {
+
+class client;
+
+namespace call_type {
```

Comment:
@R3AL great, thanks. Please bear with me as I'm going on a holiday for a few days, so I'll only be able to respond then.